core/blockchain.go

```diff
@@ -428,7 +428,20 @@ func NewBlockChain(
 	// Start tx indexer/unindexer if required.
 	if bc.cacheConfig.TxLookupLimit != 0 {
 		bc.wg.Add(1)
-		go bc.maintainTxIndex()
+		go func() {
+			defer bc.wg.Done()
+			var (
+				headCh = make(chan ChainEvent, 1) // Buffered to avoid locking up the event feed
+				sub    = bc.SubscribeChainAcceptedEvent(headCh)
+			)
+			if sub == nil {
+				log.Warn("could not create chain accepted subscription to unindex txs")
+				return
+			}
+			defer sub.Unsubscribe()
+
+			bc.maintainTxIndex(headCh)
```

Comment:
why did we move the subscription out?

Reply:
if not, it is possible (but unlikely) that the chain will accept the 1st block before the call to `chain.Accept`.
Actually I think these should be moved out of the goroutine as well